doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,889 @@
+        /// Install percentage if known.
+        Single InstallPercentage;
+    };
+
+    /// Result of the install
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass InstallResult
+    {
+        /// Used by a caller to correlate the install with a caller's data.
+        String CorrelationData{ get; };
+        /// Whether a restart is required to complete the install.
+        Boolean RebootRequired { get; };
+    }
+
+    /// IMPLEMENTATION NOTE: SourceOrigin from AppInstallerRepositorySource.h
+    /// Defines the origin of the app catalog details.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    enum AppCatalogOrigin
+    {
+        /// Predefined means it came as part of the Windows Package Manager package and cannot be removed.
+        Predefined,
+        /// User means it was added by the user and could be removed.
+        User,
+    };
+
+    /// IMPLEMENTATION NOTE: SourceTrustLevel from AppInstallerRepositorySource.h
+    /// Defines the trust level of the app catalog.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    enum AppCatalogTrustLevel
+    {
+        None,
+        Trusted,
+    };
+
+    /// IMPLEMENTATION NOTE: SourceDetails from AppInstallerRepositorySource.h
+    /// Interface for retrieving information about an app catalog without acting on it.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass AppCatalogInfo
+    {
+        /// The app catalog's unique identifier.
+        String Id { get; };
+        /// The name of the app catalog.
+        String Name { get; };
+        /// The type of the app catalog.
+        String Type { get; };
+        /// The argument used when adding the app catalog.
+        String Arg { get; };
+        /// The app catalog's extra data string.
+        String ExtraData { get; };
+        /// The last time that this app catalog was updated.
+        Windows.Foundation.DateTime LastUpdateTime { get; };
+        /// The origin of the app catalog.
+        AppCatalogOrigin Origin { get; };
+        /// The trust level of the app catalog
+        AppCatalogTrustLevel TrustLevel { get; };
+    };
+
+    /// A metadata item of a package version.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    enum PackageVersionMetadata
+    {
+        /// The InstallerType of an installed package
+        InstalledType,
+        /// The Scope of an installed package
+        InstalledScope,
+        /// The system path where the package is installed
+        InstalledLocation,
+        /// The standard uninstall command; which may be interactive
+        StandardUninstallCommand,
+        /// An uninstall command that should be non-interactive
+        SilentUninstallCommand,
+        /// The publisher of the package
+        Publisher,
+        /// The locale of the package

```

Comment:
Documentation examples use "sr-ME" (BCP47 recommended uppercase region). Documentation notes "Language must follow IETF language tag guidelines"

Reply:
FWIW, use of locale ID (LCIDs) have been discouraged since Vista.  Most of the "locales" in Windows do not even have LCIDs anymore.

Typically software is localized to languages, and a user may choose which language variant they want to use in their region.  For example, I may choose to use en-US or en-GB even if my region is CA.  It's less likely that there's a specific en-CA variant as they usually get pushed to one of the other dialects.